### PR TITLE
Make ComposteAlgorithm a struct

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
So it's still not thread-safe?

**Comment by YingboMa** [comment]
I am going to give it a try tomorrow.

File: src/solve.jl [hunk 1]
```diff
@@ -35,8 +35,8 @@ function init{algType<:OrdinaryDiffEqAlgorithm,recompile_flag}(
   qsteady_max = qsteady_min_default(alg),
   qoldinit=1//10^4, fullnormalize=true,
   failfactor = 2,
-  beta2=beta2_default(alg),
-  beta1=beta1_default(alg,beta2),
+  beta2=nothing,
```
Comment (ChrisRackauckas) [inline src/solve.jl]: does this even work?

File: src/solve.jl [hunk 2]
```diff
@@ -247,10 +227,34 @@ function init{algType<:OrdinaryDiffEqAlgorithm,recompile_flag}(
 
   if typeof(alg) <: OrdinaryDiffEqCompositeAlgorithm
```
Comment (ChrisRackauckas) [inline src/solve.jl]: oh got it

**Comment by ChrisRackauckas** [comment]
LGTM if the tests end up passing.

**Comment by ChrisRackauckas** [comment]
A similar change probably needs to be done on stochastic.

**Pull request merged**